src/recensio/plone/browser/authorsearch.py

```diff
@@ -0,0 +1,125 @@
+                "b_size": int(self.request.get("b_size", 30)),
+                "sort_on": "sortable_title",
+                "fl": "Title",
+            }
+            navigation_root = getNavigationRoot(self.context)
+            if self.request.get("use_navigation_root", True):
+                query["path"] = navigation_root
+            num_authors = len(catalog(query))
+            if num_authors == 0:
+                return 0
+            b_size = 30
+            query["b_size"] = b_size + 1
+            partition = [0, num_authors]
+            sample_idx = 0
+            while True:
+                sample_start = max(
+                    partition[0] + (partition[1] - partition[0]) / 2,
+                    0,
+                )
+                sample_start = int(sample_start / b_size) * b_size
+                query["b_start"] = max(sample_start - 1, 0)
+                sample = catalog(query)
+                sample_end = min(sample_start + b_size - 1, num_authors - 1)
+                if sample[sample_end]["Title"][0].lower() < letter:
+                    # last item in sample is before letter - move further towards end
+                    partition[0] = min(sample_end + 1, num_authors)
+                elif (
+                    sample_start == 0
+                    or sample[sample_start - 1]["Title"][0].lower() < letter
+                ):
+                    break
+                else:
+                    # first item in sample is our letter or after - move further towards beginning
+                    partition[1] = max(sample_start - 1, 0)
+
+                sample_idx += 1
+                if sample_idx >= 9:
+                    logger.warning("Bailing out to avoid infinite loop")
+                    break
+
+            b_start = sample_start
+            for idx in range(sample_start, sample_end):
+                if sample[idx]["Title"].lower().startswith(letter):
+                    b_start = idx
+                    break
+            b_start = int(b_start / b_size) * b_size
+        return b_start
+
+    @property
+    @instance.memoize
+    def authors(self):
+        catalog = getToolByName(self.context, "portal_catalog")
+        author_string = safe_unicode(self.request.get("authors"))
+        b_start = self.get_b_start()
+        b_size = int(self.request.get("b_size", 30))
+        query = {
+            "portal_type": "Person",
+            "b_start": b_start,
+            "b_size": b_size,
+            "sort_on": "sortable_title",
+            "fl": "Title,UID,path_string",
+        }
+        navigation_root = getNavigationRoot(self.context)
+        if self.request.get("use_navigation_root", True):
+            query["path"] = navigation_root
+        if author_string:
+            query["SearchableText"] = author_string.strip("\"'")
+        return catalog(query)
+
+    @property
+    def portal_title(self):
+        return getToolByName(self.context, "portal_url").getPortalObject().Title()
```

Comment:
I think it should be fetched from the registry on these days

Reply:
The registry? I would maybe use plone.api instead, I don't see how the registry is involved.

Reply:
See: 
![image](https://user-images.githubusercontent.com/1300763/216547748-25f9b388-a4e5-4570-a57c-d8b9c9f4681e.png)
and:
https://6-classic.demo.plone.org/portal_registry/edit/plone.site_title

Reply:
Interesting, thanks